File: Assignment_3/Assignment3_5.py
"""-------------------------------------------------------------------------------------------
                          Assignment3_5
                    (Student name - Vaishali Jorwekar)
                     Python by Marvellous Infosystems
---------------------------------------------------------------------------------------------
Problem statement : This program accepets N elements from the user and 
                    store it into a list
                    2. It finds prime number from the list
                    3. Returns addition of all prime numbers
------------------------------------------------------------------------------------------"""
from MarvellousFun import ChkPrime,acceptElements
#-------------------------------------------------------------------------------------
# This Function finds the prime number in the list
#-------------------------------------------------------------------------------------
def listPrime(numberList):
    sumPrime = 0
    listPrime=[]
    for num in numberList: 
         if (num>1) and ChkPrime(num) :
             listPrime.append(num)
             sumPrime+=num
    print("-------------------------------------------------------------------")
    print("Original list :",numberList)
    print("Prime Number list:",listPrime)
    print(f"Addition of Prime numbers in the list is : {sumPrime}")     
    print("-------------------------------------------------------------------")

#------------------------------------------------------------------------------------------
# This function accepts a number of elements number from user and calls function 
# acceptElements(number), listPrime(inputNumberList)
#------------------------------------------------------------------------------------------
def main():
    noOfElements=int(input("Enter number of elements in the list:"))
    inputNumberList=acceptElements(noOfElements)
    listPrime(inputNumberList)
#-----------------------------------------------------------------------------------
# This is main entry point of the program
#-----------------------------------------------------------------------------------
if __name__=="__main__":
    main()
#-----------------------------------------------------------------------------------

File: JavaPractice/Assignment_3/Assignment3_5.java
/*-------------------------------------------------------------------------------------------
                          Assignment3_5
                    (Student name - Vaishali Jorwekar)
                     Python by Marvellous Infosystems
---------------------------------------------------------------------------------------------
Problem statement : This program accepets N elements from the user and 
                    store it into a list
                    2. It finds prime number from the list
                    3. Returns addition of all prime numbers
------------------------------------------------------------------------------------------*/


import java.util.ArrayList;
import java.util.Scanner;




public class Assignment3_5 {

	public static void main(String args[]) {

		ArrayList<Integer> numberList=Module.acceptNumbers();
		System.out.println("List Elements are:"+numberList);

		sumPrimeNumbers(numberList);



	}

	public static boolean checkPrime(int number) {
		boolean isPrime=true;
		if(number<=1) {
			isPrime= false;
		}
		for(int cnt=2;cnt<number;cnt++) {
			if(number%cnt==0) {
				isPrime= false;
				break;
			}
		}
		return isPrime;
	}


	
	private static void sumPrimeNumbers(ArrayList<Integer> numberList) {
		int sumPrime=0;
		StringBuffer primeList=new StringBuffer();
		for(int cnt=0;cnt<(numberList.size());cnt++) {
			int number=numberList.get(cnt);
			if(checkPrime(number)) {
				primeList.append(number);
				primeList.append(" ");
				sumPrime=sumPrime+number;
			}
		}	
		System.out.println("Sum of "+primeList+" Prime Numbers is :"+sumPrime);
	}

}
